core/node/external_proof_integration_api/src/processor.rs

```diff
@@ -93,38 +124,57 @@ impl Processor {
             .get_latest_proven_batch()
             .await?;
 
-        let l1_batch_number = if let Some(l1_batch_number) = request.0 .0 {
-            if l1_batch_number > latest_available_batch {
-                tracing::error!(
-                    "Requested batch is not available: {:?}, latest available batch is {:?}",
-                    l1_batch_number,
-                    latest_available_batch
-                );
-                return Err(ProcessorError::BatchNotReady(l1_batch_number));
-            }
+        let proof_generation_data = self
+            .proof_generation_data_for_existing_batch_internal(latest_available_batch)
+            .await;
+
+        match proof_generation_data {
+            Ok(data) => Ok(ProofGenerationDataResponse(data)),
+            Err(err) => Err(err),
+        }
+    }
+
+    #[tracing::instrument(skip(self))]
```

Comment:
Nit: Why do we need tracing here if we have metrics? Given that we don't sample traces, we don't want to record traces of user-triggered calls.

I understand that it wasn't introduced here, just noticed.